<launch>
  <arg name="object_name" default="bluebird" />
  <arg name="capability_group" default="Vicon" />
  <node ns="$(arg object_name)" name="vrpn_client" type="ros_vrpn_client" pkg="ros_vrpn_client" output="screen">
    <param name="vrpn_server_ip" value="vicon" />
    <param name="vrpn_coordinate_system" value="vicon" />
    <param name="timestamping_system" value="ros" />
    <param name="object_name" value="$(arg object_name)" />
    <param name="verbose" value="true" />
    <param name="translational_estimator/kp" value="0.5" />
    <param name="translational_estimator/kv" value="0.5" />
    <param name="rotational_estimator/orientation_estimate_initial_covariance" value="100.0" />
    <param name="rotational_estimator/rate_estimate_initial_covariance" value="100.0" />
    <param name="rotational_estimator/orientation_process_covariance" value="0.000002" />
    <param name="rotational_estimator/rate_process_covariance" value="10.0" />
    <param name="rotational_estimator/orientation_measurementCovariance" value="0.001" />
    <param name="rotational_estimator/outlier_rejection_method" value="mahalanobis_distance" />
    <param name="rotational_estimator/outlier_rejection_mahalanobis_threshold" value="4.0" />
    <param name="rotational_estimator/outlier_rejection_subsequent_threshold_degrees" value="30.0" />
    <param name="rotational_estimator/outlier_rejection_subsequent_maximum_count" value="50.0" />
    <param name="rotational_estimator/output_minimal_quaternions" value="false" />
    <param name="rotational_estimator/output_minimal_quaternions" value="false" />
    <param name="capability_group" value="$(arg capability_group)" />

    <!-- Rotation of the vicon frame in the imu frame-->
    <param name="q_vi_w" value="0.999828" />
    <param name="q_vi_x" value="0.0167064" />
    <param name="q_vi_y" value="-0.007213" />
    <param name="q_vi_z" value="-0.003516" />
    <param name="t_vi_x" value="0.0" />
    <param name="t_vi_y" value="0.0" />
    <param name="t_vi_z" value="0.0" />

  </node>
</launch>
